src/main/java/frc/robot/swerve/ModuleIOReal.java

```diff
@@ -0,0 +1,181 @@
+    private double currentAngle;
+    private double driveMotorVelocitySetpoint;
+
+    public ModuleIOReal(int driveMotorID, int angleMotorID, int encoderID,
+                        double[] motionMagicConfigs, int number) {
+        this.driveMotor = new TalonFX(driveMotorID);
+        this.angleMotor = new TalonFX(angleMotorID);
+
+        this.encoder = new DutyCycleEncoder(encoderID);
+
+        TalonFXConfiguration driveMotorConfig = new TalonFXConfiguration();
+        Slot0Configs drivePIDGains = new Slot0Configs()
+                .withKP(SwerveConstants.DRIVE_kP)
+                .withKI(SwerveConstants.DRIVE_kI)
+                .withKD(SwerveConstants.DRIVE_kD)
+                .withKV(SwerveConstants.DRIVE_KF);
+
+        driveMotorConfig.Slot0 = drivePIDGains;
+        driveMotorConfig.Voltage.PeakForwardVoltage = SwerveConstants.VOLT_COMP_SATURATION;
+        driveMotorConfig.Voltage.PeakReverseVoltage = -SwerveConstants.VOLT_COMP_SATURATION;
+        driveMotorConfig.CurrentLimits.SupplyCurrentLimit = SwerveConstants.SUPPLY_CURRENT_LIMIT.currentLimit;
+        driveMotorConfig.CurrentLimits.StatorCurrentLimit = SwerveConstants.STATOR_CURRENT_LIMIT.currentLimit;
+        driveMotorConfig.ClosedLoopGeneral.ContinuousWrap = true;
+        driveMotorConfig.Feedback.SensorToMechanismRatio = 1 / SwerveConstants.DRIVE_REDUCTION;
+        driveMotorConfig.Feedback.RotorToSensorRatio = 1;
+        driveMotor.getConfigurator().apply(driveMotorConfig);
+
+        driveMotor.setNeutralMode(NeutralModeValue.Brake);
+        driveMotor.setInverted(true);
+
+        TalonFXConfiguration angleMotorConfig = new TalonFXConfiguration();
+        Slot0Configs anglePIDGains = new Slot0Configs()
+                .withKP(motionMagicConfigs[0])
+                .withKI(motionMagicConfigs[1])
+                .withKD(motionMagicConfigs[2])
+                .withKV(motionMagicConfigs[3])
+                .withKS(motionMagicConfigs[4]);
+
+        angleMotorConfig.Slot0 = anglePIDGains;
+        angleMotorConfig.MotionMagic.MotionMagicJerk = motionMagicConfigs[5];
+        angleMotorConfig.MotionMagic.MotionMagicCruiseVelocity = motionMagicConfigs[6];
+        angleMotorConfig.MotionMagic.MotionMagicAcceleration = motionMagicConfigs[7];
+        angleMotorConfig.Voltage.PeakForwardVoltage = SwerveConstants.VOLT_COMP_SATURATION;
+        angleMotorConfig.Voltage.PeakReverseVoltage = -SwerveConstants.VOLT_COMP_SATURATION;
+        angleMotorConfig.CurrentLimits.SupplyCurrentLimit = SwerveConstants.SUPPLY_CURRENT_LIMIT.currentLimit;
+        angleMotorConfig.CurrentLimits.StatorCurrentLimit = SwerveConstants.STATOR_CURRENT_LIMIT.currentLimit;
+        angleMotorConfig.ClosedLoopGeneral.ContinuousWrap = true;
+        angleMotorConfig.Feedback.SensorToMechanismRatio = 1 / SwerveConstants.ANGLE_REDUCTION;
+        angleMotorConfig.Feedback.RotorToSensorRatio = 1;
+        angleMotorConfig.MotorOutput.DutyCycleNeutralDeadband = SwerveConstants.NEUTRAL_DEADBAND;
```

Comment:
All that should be in the constants file.

Reply:
ca87e4b2668351dd858d8a56fdf9504c2622d339

Reply:
Still a few remain

Reply:
The only one that is still there is the continuous wrap which doesn't have a `.withContinuousWrap()` method so it can't be configured in the constants.

Reply:
For some reason the current newest commit (cba1570) uses Phoenix 5 classes. Interactive rebase or bisect may be useful here. 
Also the continuous wrap is discussed below.